### PR TITLE
Impl common traits

### PR DESCRIPTION
**Pull request by XOSplicer**

### CODE REVIEW

**Comment by XOSplicer** [comment]
bors try

File: src/binary_heap.rs [hunk 1]
```diff
@@ -412,6 +412,42 @@ impl<'a, T> Drop for Hole<'a, T> {
...
+    K: Kind,
+    T: Ord + Clone,
+{
+    fn clone(&self) -> Self {
+        BinaryHeap {
+            _kind: PhantomData,
+            data: self.data.clone(),
+        }
+    }
+}
```
Comment (TeXitoi) [inline src/binary_heap.rs:438]: `#[derive(Clone)]` should work

File: src/indexset.rs [hunk 2]
```diff
@@ -478,6 +478,19 @@ where
...
+where
+    K: Eq + Hash + Clone,
+    S: BuildHasher + Default,
+    N: ArrayLength<Bucket<K, ()>> + ArrayLength<Option<Pos>>,
+{
+    fn clone(&self) -> Self {
+        IndexSet {
+            map: self.map.clone()
+        }
+    }
```
Comment (TeXitoi) [inline src/indexset.rs:491]: `#[derive(Clone)]` should work

File: src/binary_heap.rs (hunk 1 above)
Comment (XOSplicer) [reply src/binary_heap.rs:438]: the `Min` and `Max` enums don't implement Clone, as they are only used as markers, therefore deriving Clone was not possible. But you are right this should work aswell, when `Min` and `Max` also derive Clone

File: src/indexset.rs (hunk 2 above)
Comment (XOSplicer) [reply src/indexset.rs:491]: The traitbounds of the impl and the derive are not the same though

File: src/binary_heap.rs (hunk 1 above)
Comment (TeXitoi) [reply src/binary_heap.rs:438]: OK.

File: src/indexset.rs (hunk 2 above)
Comment (TeXitoi) [reply src/indexset.rs:491]: OK, good catch

**Review by TeXitoi — APPROVED** [review]

**Pull request closed without merging**